diff --git a/webshop/webshop.sql b/webshop/webshop.sql
index 10db88dfd7f840abfde800ff166015bab9a4f8e8..fdabde2c4ba6eef77a99517fea485f23b4234d5d 100644
--- a/webshop/webshop.sql
+++ b/webshop/webshop.sql
@@ -51,4 +51,27 @@ VALUES
 	(17,'SKISCHOOL.CHILDS','Skischool Kinder',120.00,4,'Find your own personal way to unlimited snowboarding freedom. You\'ll soon be learning nose and tail turns or trying out the halfpipe.','media/images/skischool/childs.jpg',900),
 	(18,'SKISCHOOL.ADULTS','Skischool Adults',140.00,4,'Find your own personal way to unlimited snowboarding freedom. You\'ll soon be learning nose and tail turns or trying out the halfpipe.','media/images/skischool/adults.png',900);
 
+CREATE TABLE `order` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `created` datetime NOT NULL,
+  `prename` varchar(255) NOT NULL DEFAULT '',
+  `surname` varchar(255) NOT NULL DEFAULT '',
+  `address` text NOT NULL,
+  `zip` varchar(10) NOT NULL DEFAULT '',
+  `city` varchar(255) NOT NULL DEFAULT '',
+  `phone` varchar(255) NOT NULL DEFAULT '',
+  `email` varchar(255) NOT NULL DEFAULT '',
+  PRIMARY KEY (`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+CREATE TABLE `order_product` (
+  `order` int(11) NOT NULL,
+  `product` int(11) NOT NULL,
+  `price` decimal(12,2) NOT NULL,
+  `qty` int(11) NOT NULL DEFAULT '1',
+  PRIMARY KEY (`order`,`product`),
+  CONSTRAINT `order` FOREIGN KEY (`order`) REFERENCES `order` (`id`),
+  CONSTRAINT `product` FOREIGN KEY (`product`) REFERENCES `product` (`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
 GRANT ALL ON *.* TO webshop@localhost IDENTIFIED by 'Pear73';
